scripts/readme_toc.py

```diff
@@ -1,119 +1,92 @@
-        toc.append(f"{indent}- [{text}](#{slug})")
+        slug = PUNCT_RE.sub("", slug).strip().replace(" ", "-")
+        indent = "  " * (level - 2)
+        toc.append(f"{indent}- [{title}](#{slug})")
     return toc
 
 
-def check_or_fix(readme_path: Path, fix: bool) -> int:
-    if not readme_path.is_file():
-        print(f"Error: file not found: {readme_path}", file=sys.stderr)
+def process(path: Path, fix: bool) -> int:
+    """Check or fix the ToC in the given Markdown file."""
+    if not path.is_file():
+        print(f"Error: file not found: {path}", file=sys.stderr)
         return 1
-    content = readme_path.read_text(encoding="utf-8")
-    lines = content.splitlines()
-    # locate ToC markers
+    text = path.read_text(encoding="utf-8")
+    lines = text.splitlines()
+
+    # Locate markers
     try:
-        begin_idx = next(i for i, l in enumerate(lines) if l.strip() == BEGIN_TOC)
-        end_idx = next(i for i, l in enumerate(lines) if l.strip() == END_TOC)
+        i1 = next(i for i, l in enumerate(lines) if l.strip() == BEGIN_TOC)
+        i2 = next(i for i, l in enumerate(lines) if l.strip() == END_TOC)
     except StopIteration:
-        print(
-            f"Error: Could not locate '{BEGIN_TOC}' or '{END_TOC}' in {readme_path}.",
-            file=sys.stderr,
-        )
+        print(f"Error: missing ToC markers in {path}", file=sys.stderr)
         return 1
-    # extract current ToC list items
-    current_block = lines[begin_idx + 1 : end_idx]
-    current = [l for l in current_block if l.lstrip().startswith("- [")]
-    # generate expected ToC
-    expected = generate_toc_lines(content)
+
+    current = [l for l in lines[i1+1:i2] if l.lstrip().startswith("- [")]
+    expected = generate_toc(text)
+
     if current == expected:
         return 0
+
     if not fix:
-        print(
-            "ERROR: README ToC is out of date. Diff between existing and generated ToC:"
-        )
-        # Show full unified diff of current vs expected
-        diff = difflib.unified_diff(
-            current,
-            expected,
-            fromfile="existing ToC",
-            tofile="generated ToC",
-            lineterm="",
-        )
-        for line in diff:
-            print(line)
+        # Show diff
+        print("ERROR: ToC out of date. Diff:")
+        for ln in unified_diff(current, expected, fromfile="existing ToC", tofile="generated ToC", lineterm=""):
+            print(ln)
         return 1
-    # rebuild file with updated ToC
-    prefix = lines[: begin_idx + 1]
-    suffix = lines[end_idx:]
-    new_lines = prefix + [""] + expected + [""] + suffix
-    readme_path.write_text("\n".join(new_lines) + "\n", encoding="utf-8")
-    print(f"Updated ToC in {readme_path}.")
+
+    # Rewrite file with new ToC
+    new = lines[:i1+1] + [""] + expected + [""] + lines[i2:]
+    path.write_text("\n".join(new) + "\n", encoding="utf-8")
+    print(f"Updated ToC in {path}.")
     return 0
 
 
+def main():
```

Comment:
Please reorganize `main()` as I had it before:

- Top of file so reader does not have to dig to the end of the file to figure out what arguments this script accepts.
- As good practice, `main()` should not call `sys.exit()` itself. That behavior should be restricted to `if __name__ == "__main__"`